Tests/OpenCombineTests/PublisherTests/ReplaceNilTests.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  ReplaceNilTests.swift
+//  
+//
+//  Created by Joseph Spadafora on 7/4/19.
+//
+
+import XCTest
+
+#if OPENCOMBINE_COMPATIBILITY_TEST
+import Combine
+#else
+import OpenCombine
+#endif
+
+@available(macOS 10.15, *)
+final class ReplaceNilTests: XCTestCase {
+    static let allTests = [
+        ("testReturnsMapPublisher", testReturnsMapPublisher),
+        ("testReplacesNilElement", testReplacesNilElement),
+        ("testExistingElementIsPreserved", testExistingElementIsPreserved)
+    ]
+
+    func testReturnsMapPublisher() {
+        // Given
+        let nilPublisher = PassthroughSubject<Int?, Never>()
+
+        // When
+        let replaceNil = nilPublisher.replaceNil(with: 42)
+
+        // Then
+        let mirror = Mirror(reflecting: replaceNil)
+        let expected = "Mirror for Map<PassthroughSubject<Optional<Int>, Never>, Int>"
+        XCTAssertEqual(mirror.description, expected)
+    }
+
+    func testReplacesNilElement() {
+        // Given
+        let nilPublisher = PassthroughSubject<Int?, Never>()
+        let output = Int.random(in: 1...100)
+        let sut = nilPublisher.replaceNil(with: output)
+        let subscriber = TrackingSubscriberBase<Int, Never>(
+            receiveSubscription: {
+                $0.request(.unlimited)
+            }
+        )
+
+        // When
+        sut.receive(subscriber: subscriber)
+        nilPublisher.send(nil)
+
+        // Then
+        XCTAssertEqual(subscriber.history, [.subscription(Subscriptions.empty),
+                                            .value(output)])
+    }
+
+    func testExistingElementIsPreserved() {
+        // Given
+        let nilPublisher = PassthroughSubject<Int?, Never>()
+        let output = Int.random(in: 1...100)
+        let sut = nilPublisher.replaceNil(with: output + 1)
+        let subscriber = TrackingSubscriberBase<Int, Never>(
+            receiveSubscription: {
+                $0.request(.unlimited)
+            }
+        )
+
+        // When
+        sut.receive(subscriber: subscriber)
+        nilPublisher.send(output)
+
+        // Then
+        XCTAssertEqual(subscriber.history, [.subscription(Subscriptions.empty),
+                                            .value(output)])
+    }
```

Comment:
I suggest to combine the above two tests into one test, where multiple values are sent. Something like:

```swift
testReplacesNilElement() {
    // Given
    let passthrough = PassthroughSubject<Int?, Never>()
    let sut = passthrough.replaceNil(with: 42)
    let subscriber = TrackingSubscriberBase<Int, Never>(
        receiveSubscription: {
            $0.request(.unlimited)
        }
    )

    // When
    sut.subscribe(subscriber)
    passthrough.send(1)
    passthrough.send(2)
    passthrough.send(nil)
    passthrough.send(4)
    passthrough.send(5)
    passthrough.send(nil)
    passthrough.send(completion: .finished)

    // Then
    XCTAssertEqual(subscriber.history, [.subscription("PassthroughSubject"),
                                        .value(1),
                                        .value(2),
                                        .value(42),
                                        .value(4),
                                        .value(5),
                                        .value(42),
                                        .completion(.finished)])
}
```

Since this operator trivially uses `Map` under the hood (which already has substantial test coverage), I believe we can be a little lazier when it comes to testing it, and just test the basic behavior.

Reply:
I think it makes sense to add another test, but it's a good practice to keep tests small and focused.  The other two tests very simply test the other two paths and would help us find any issues.